src/client/renderer/r_bsp_model.c

```diff
@@ -667,11 +667,11 @@ static void R_LoadBspPlanes(r_bsp_model_t *bsp, const d_bsp_lump_t *l) {
 
 		out->dist = LittleFloat(in->dist);
 		out->type = LittleLong(in->type);
-		out->sign_bits = Cm_SignBitsForPlane(out);
+		out->sign_bits = Cm_SignBitsForPlane((cm_bsp_plane_t *) out);
 		out->num = (i >> 1) + 1;
 	}
 
-	r_shadow_state.plane_shadow_counts = Mem_LinkMalloc(((count >> 1) + 1) * sizeof(int16_t), bsp);
+	bsp->plane_shadows = Mem_LinkMalloc(((count >> 1) + 1) * sizeof(uint16_t), bsp);
 }
```

Comment:
This can live inside of the BSP again, now that the memory corruption has been tracked down and eliminated.